x/authz/module/abci.go

```diff
@@ -0,0 +1,20 @@
+package authz
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz/keeper"
+)
+
+// EndBlocker is called at the end of every block
+func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+
+	matureGrants, err := keeper.DequeueAllMatureGrants(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	// clears all the mature grants
+	if err := keeper.DeleteExpiredGrants(ctx, matureGrants); err != nil {
+		panic(err)
+	}
```

Comment:
We don't need 2 separate functions: `DequeueAllMatureGrants` and `DeleteExpiredGrants`. They can be merged: deleting grant can be done here: https://github.com/cosmos/cosmos-sdk/pull/10714/files#diff-770a5c3d50c8cb65512e55d43e076ced18a8627e809ccd01adedcb8774ab721dR386